src/server/repositories/LinkStatisticsRepository.ts

```diff
@@ -0,0 +1,74 @@
+import { injectable } from 'inversify'
+import { Url, UrlType } from '../models/url'
+import { Clicks, ClicksType } from '../models/statistics/daily'
+import { Devices, DevicesType } from '../models/statistics/devices'
+import { WeekdayClicks, WeekdayClicksType } from '../models/statistics/weekday'
+import {
+  DailyClicksInterface,
+  DeviceClicksInterface,
+  LinkStatisticsInterface,
+  WeekdayClicksInterface,
+} from '../../shared/interfaces/link-statistics'
+import { LinkStatisticsRepositoryInterface } from './interfaces/LinkStatisticsRepositoryInterface'
+
+export type UrlStats = UrlType & {
+  DeviceClicks?: DevicesType
+  DailyClicks: ClicksType[]
+  WeekdayClicks: WeekdayClicksType[]
+}
+
+@injectable()
+export class LinkStatisticsRepository
+  implements LinkStatisticsRepositoryInterface {
+  /**
+   * Retrieves link statistics for a specified short link.
+   *
+   * @param shortUrl The target short url to retrieve link statistics.
+   */
+  public findByShortUrl: (
+    shortUrl: string,
+  ) => Promise<LinkStatisticsInterface | null> = async (shortUrl) => {
+    const url = await Url.findOne({
+      where: { shortUrl },
+      include: [
+        { model: Devices, as: 'DeviceClicks' },
+        { model: Clicks, as: 'DailyClicks' },
```

Comment:
i believe we also need to filter `DailyClicks` by date range too?

Reply:
I have added that, and also ordering or daily clicks and weekday clicks. I hope to get some comments from you regarding my [implementation](https://github.com/opengovsg/GoGovSG/blob/stats/src/server/repositories/LinkStatisticsRepository.ts)